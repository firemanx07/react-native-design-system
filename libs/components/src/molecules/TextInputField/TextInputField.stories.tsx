import { ComponentMeta, ComponentStory } from '@storybook/react';
import React from 'react';

import {
  StorybookRow,
  StorybookScreen,
  StorybookSection,
} from '../../storybook';
import { default as TextInputFieldBase, TextProps } from './TextInputField';

const Story: ComponentMeta<typeof TextInputFieldBase> = {
  component: TextInputFieldBase,
  title: 'molecules/TextInputField',
};
export default Story;

const noop = () => {
  // This is a mock
};
const Template: ComponentStory<typeof TextInputFieldBase> = args => {
  return (
    <StorybookScreen title="Text Input Field:">
      <StorybookSection>
        <TextInputFieldBase {...args} />
      </StorybookSection>
      <StorybookRow>
        <StorybookSection title="Disabled input field with text:">
          <TextInputFieldBase
            {...args}
            value="Value"
            editable={false}
            isError={false}
            onChangeText={noop}
          />
        </StorybookSection>
        <StorybookSection title="Success input field with text:">
          <TextInputFieldBase {...args} value="Success Value" isError={false} />
        </StorybookSection>
        <StorybookSection title="Error input field with text:">
          <TextInputFieldBase
            {...args}
            isError
            errorMessage="Oops! Somethings not right here."
          />
        </StorybookSection>
      </StorybookRow>
    </StorybookScreen>
  );
};

const parameters = {
  controls: {
    include: [
      'placeholder',
      'value',
      'isValid',
      'isError',
      'editable',
      'errorMessage',
    ],
  },
};
const defaultArgs: TextProps = {
  onChangeText: noop,
  placeholder: 'Placeholder',
  value: '',
  editable: true,
  isError: false,
  errorMessage: 'Oops ! Somethings not right here.',
};
export const TextInputField = Template.bind({});
TextInputField.parameters = parameters;
TextInputField.args = {
  ...defaultArgs,
};
