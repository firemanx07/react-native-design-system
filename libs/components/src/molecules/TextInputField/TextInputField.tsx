import { IconSizeType, styled, useNamespacedTheme } from '@proxym/themes';
import { rgba } from 'polished';
import React, {
  memo,
  ReactNode,
  useCallback,
  useMemo,
  useRef,
  useState,
} from 'react';
import {
  NativeSyntheticEvent,
  Platform,
  StyleProp,
  TextInputFocusEventData,
  TextInputProps,
  TextStyle,
  ViewStyle,
} from 'react-native';
import * as ReactNative from 'react-native';

import { BaseText, TextVariant, TextWeight } from '../../atoms';
import { TestIDType } from '../../types';

export type AccessoryRenderFunction = (
  size: number,
  color: string,
) => ReactNode;
export type TextInputAccessoryProps = {
  leftAccessory?: AccessoryRenderFunction;
  rightAccessory?: AccessoryRenderFunction;
};
export type BaseTextInputProps = TextInputProps & {
  value: string;
  placeholder?: string;
  onChangeText?: (text: string) => void;
  autoFocus?: boolean;
  editable?: boolean;
  style?: {
    container?: StyleProp<ViewStyle>;
    text: StyleProp<TextStyle>;
    shadow?: StyleProp<ViewStyle>;
  };
  onFocus?: () => void;
  errorMessage?: string;
  isError?: boolean;
  isSuccess?: boolean;
  errorTextStyle?: StyleProp<TextStyle>;
  label?: string;
  width?: number;
  isPicker?: boolean;
  onPress?: () => void;
  disabled?: boolean;
  pointerEvents?: string;
} & TextInputAccessoryProps &
  TestIDType;

const TEXT_INPUT_HEIGHT = 48;
const TEXT_MULTILINE_HEIGHT = 121;
const TextInputField = ({
  onFocus,
  width = 300,
  onBlur,
  onChangeText,
  isError,
  isSuccess,
  value,
  label,
  editable = true,
  placeholder,
  autoFocus,
  keyboardType,
  errorMessage,
  errorTextStyle,
  style,
  isPicker = false,
  onPress,
  disabled,
  pointerEvents,
  onLayout,
  ...rest
}: BaseTextInputProps) => {
  const [isFocused, setIsFocused] = useState(false);
  const { colors } = useNamespacedTheme();
  const textInputRef = useRef<ReactNative.TextInput>(null);
  const handleFocus = useCallback(() => {
    onFocus && onFocus();
    setIsFocused(true);
  }, [onFocus]);

  const handlePress = useCallback(() => {
    textInputRef?.current?.focus();
    onPress && onPress();
  }, [onPress]);

  const handleBlur = useCallback<
    (e: NativeSyntheticEvent<TextInputFocusEventData>) => void
  >(
    e => {
      onBlur && onBlur(e);
      setIsFocused(false);
    },
    [onBlur],
  );
  const borderColor = useMemo(() => {
    if (isError) {
      return colors.error;
    }
    if (isFocused && !isPicker) {
      return colors.secondary;
    }
    if (isSuccess) {
      return colors.success;
    }
    return colors.grayDark;
  }, [isError, isFocused, isSuccess, colors]);
  const accessoryColor = useMemo(() => {
    if (!editable) return colors.grayDark;
    return borderColor === colors.grayDark ? colors.dark : borderColor;
  }, [borderColor, editable, colors.grayDark, colors.dark]);
  return (
    <Container onLayout={onLayout}>
      {label && (
        <BaseText variant={TextVariant.S} weight={TextWeight.Medium}>
          {label}
        </BaseText>
      )}
      <TextInputContainer
        editable={editable}
        width={width}
        color={rgba(borderColor, 0.5)}
        onPress={handlePress}
        disabled={disabled}
      >
        {rest.leftAccessory &&
          rest.leftAccessory(IconSizeType.primary, accessoryColor)}
        <TextInput
          ref={textInputRef}
          {...rest}
          value={value}
          placeholder={placeholder}
          placeholderTextColor={rgba(colors.dark, 0.5)}
          onChangeText={val => onChangeText && onChangeText(val)}
          onFocus={handleFocus}
          onBlur={handleBlur}
          editable={editable && !isPicker}
          autoFocus={Platform.OS === 'ios' ? autoFocus ?? false : false}
          keyboardType={keyboardType}
        />
        {rest.rightAccessory &&
          rest.rightAccessory(IconSizeType.primary, accessoryColor)}
      </TextInputContainer>
      {isError && (
        <ErrorText style={errorTextStyle} variant={TextVariant.S}>
          {errorMessage}
        </ErrorText>
      )}
    </Container>
  );
};

type CommonProps = {
  color: string;

  editable?: boolean;
  width: number;
};
const Container = styled.View`
  align-items: flex-start;
  justify-content: flex-start;
`;

const TextInputContainer = styled.Pressable<CommonProps>`
  width: ${({ width }) => width}px;
  flex-direction: row;
  /* Common.textInputContainer */
  border-radius: 7px;
  border-width: 1px;
  border-color: ${({ color, editable }) =>
    (editable && color) || 'transparent'};
  background-color: ${({ theme, editable }) =>
    editable ? theme.ds.colors.light : rgba(theme.ds.colors.grayDark, 0.2)};
  min-height: ${TEXT_INPUT_HEIGHT}px;
  padding-horizontal: ${({ theme }) => theme.ds.spacing.small}px;
  /* Layout.rowHCenter */
  align-items: center;
  /* Layout.justifyContentBetween */
  justify-content: space-between;
`;

const TextInput = styled.TextInput<TextInputAccessoryProps>`
  font-family: ${({ theme, value }) =>
    theme.ds.fonts.family[value ? 'primaryMedium' : 'primary']};
  font-size: ${({ theme }) => theme.ds.fonts.size.medium}px;
  color: ${({ theme }) => theme.ds.colors.dark};
  width: 100%;
  margin-left: ${({ theme }) => theme.ds.spacing.small}px;
  height: ${({ multiline }) =>
    multiline ? TEXT_MULTILINE_HEIGHT : TEXT_INPUT_HEIGHT}px;
  ${({ multiline, theme }) =>
    multiline && 'margin-vertical:' + theme.ds.spacing.small + 'px;'}
  ${({ rightAccessory, theme }) =>
    rightAccessory && 'margin-right:' + theme.ds.spacing.tiny + 'px;'}
`;

const ErrorText = styled(BaseText)`
  color: ${props => props.theme.ds.colors.error};
  margin-top: ${props => props.theme.ds.spacing.tiny}px;
`;

export default memo(TextInputField);
