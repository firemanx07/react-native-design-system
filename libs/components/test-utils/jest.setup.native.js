import { Animated } from 'react-native';

require('react-native-reanimated/lib/reanimated2/jestUtils').setUpTests();
require('react-native-gesture-handler/jestSetup.js');

console.warn = () => {
  //this is mock
};

Animated['timing'] = (value, config) => ({
  start: callback => {
    value.setValue(config.toValue);
    callback && callback({ finished: true });
  },
  stop: jest.fn(),
});
