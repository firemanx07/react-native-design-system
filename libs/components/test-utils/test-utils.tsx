import { theme, ThemeProvider } from '@celebrate-app/themes';
import { Story } from '@storybook/react';
import {
  cleanup,
  render,
  RenderOptions,
  waitFor,
} from '@testing-library/react-native';
import React, { ReactNode } from 'react';

const Providers = ({ children }: { children: ReactNode }) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

export const customRender = (
  ui: React.ReactElement<any>,
  options?: RenderOptions
) => {
  return render(ui, { wrapper: Providers, ...options });
};

beforeEach(cleanup);
afterEach(cleanup);

export function testStory(displayName: string, TestComponent: Story) {
  test(`${displayName} snapshot test`, async () => {
    const comp = customRender(<TestComponent {...TestComponent.args} />);
    jest.runOnlyPendingTimers();

    await waitFor(() => {
      expect(comp).toBeTruthy();
      expect(comp.toJSON()).toMatchSnapshot();
    });
  });
}

export function testComponentSnapshot(
  TestComponent: React.ReactElement<
    any,
    string | React.JSXElementConstructor<any>
  >
) {
  test('Snapshot test', () => {
    const comp = customRender(TestComponent);
    expect(comp).toBeTruthy();
    expect(comp.toJSON()).toMatchSnapshot();
  });
}

export function testAllStories(stories: any) {
  Object.keys(stories).forEach((story) => {
    if (story !== 'default') {
      testStory(story, stories[story]);
    }
  });
}
