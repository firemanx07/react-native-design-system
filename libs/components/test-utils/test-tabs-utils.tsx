import { waitFor } from '@testing-library/react-native';

import { customRender } from './test-utils';

type renderTabsProps = {
  elem: JSX.Element;
  testIDs: string[];
};
export const renderTabs = ({ elem, testIDs }: renderTabsProps) => {
  const comp = customRender(elem);
  const tabs: ReturnType<typeof comp.getByTestId>[] = [];
  testIDs.forEach(testID => tabs.push(comp.getByTestId(testID)));
  return { comp, tabs };
};
export const testTabs = async (args: ReturnType<typeof renderTabs>) => {
  await waitFor(() => {
    expect(args.comp).toBeTruthy();
    args.tabs.forEach(tab => expect(tab).toBeDefined());
  });
};
