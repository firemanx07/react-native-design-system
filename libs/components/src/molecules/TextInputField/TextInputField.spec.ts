import { testAllStories } from '../../../test-utils';
import * as stories from './TextInputField.stories';

describe('Slider tests', () => {
  testAllStories(stories);
});
