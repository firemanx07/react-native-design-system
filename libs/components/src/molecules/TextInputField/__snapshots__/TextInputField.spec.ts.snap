// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider tests TextInputField snapshot test 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "minHeight": 600,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
    ]
  }
>
  <Text
    fontFamily="Rubik-Bold"
    fontSize={20}
    lineHeight={26}
    style={
      [
        {
          "color": "#2C2C2C",
          "fontFamily": "Rubik-Bold",
          "fontSize": 20,
          "lineHeight": 26,
        },
        {
          "marginBottom": 24,
        },
      ]
    }
  >
    Text Input Field:
  </Text>
  <View
    style={
      [
        {
          "borderColor": "#2C2C2C",
          "borderRadius": 4,
          "borderStyle": "dashed",
          "borderWidth": 1,
          "marginBottom": 24,
          "marginLeft": 24,
          "marginRight": 24,
          "marginTop": 24,
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": 370,
        },
      ]
    }
    width={370}
  >
    <View
      style={
        [
          {
            "alignItems": "flex-start",
            "justifyContent": "flex-start",
          },
        ]
      }
    >
      <View
        color="rgba(199,199,210,0.5)"
        editable={true}
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderColor": "rgba(199,199,210,0.5)",
              "borderRadius": 7,
              "borderWidth": 1,
              "justifyContent": "space-between",
              "minHeight": 48,
              "paddingHorizontal": 8,
              "width": 300,
            },
          ]
        }
        width={300}
      >
        <TextInput
          autoFocus={false}
          editable={true}
          multiline={false}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="Placeholder"
          placeholderTextColor="rgba(44,44,44,0.5)"
          style={
            [
              {
                "color": "#2C2C2C",
                "fontFamily": "Rubik-Regular",
                "fontSize": 16,
                "height": 48,
                "marginLeft": 8,
                "width": "100%",
              },
            ]
          }
          value=""
        />
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "borderColor": "#2C2C2C",
            "borderRadius": 4,
            "borderStyle": "dashed",
            "borderWidth": 1,
            "marginBottom": 24,
            "marginLeft": 24,
            "marginRight": 24,
            "marginTop": 24,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": 370,
          },
        ]
      }
      width={370}
    >
      <Text
        fontFamily="Rubik-Bold"
        fontSize={16}
        lineHeight={20}
        style={
          [
            {
              "color": "#2C2C2C",
              "fontFamily": "Rubik-Bold",
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "backgroundColor": "#DEDFE0",
              "left": 16,
              "paddingHorizontal": 8,
              "position": "absolute",
              "top": -12,
            },
          ]
        }
      >
        Disabled input field with text:
      </Text>
      <View
        style={
          [
            {
              "alignItems": "flex-start",
              "justifyContent": "flex-start",
            },
          ]
        }
      >
        <Text
          fontFamily="Rubik-Medium"
          fontSize={12}
          lineHeight={18}
          style={
            [
              {
                "color": "#2C2C2C",
                "fontFamily": "Rubik-Medium",
                "fontSize": 12,
                "lineHeight": 18,
              },
            ]
          }
        >
          disabled Text input
        </Text>
        <View
          color="rgba(199,199,210,0.5)"
          editable={false}
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "rgba(199,199,210,0.2)",
                "borderColor": "transparent",
                "borderRadius": 7,
                "borderWidth": 1,
                "justifyContent": "space-between",
                "minHeight": 48,
                "paddingHorizontal": 8,
                "width": 300,
              },
            ]
          }
          width={300}
        >
          <TextInput
            autoFocus={false}
            editable={false}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder="Placeholder"
            placeholderTextColor="rgba(44,44,44,0.5)"
            style={
              [
                {
                  "color": "#2C2C2C",
                  "fontFamily": "Rubik-Medium",
                  "fontSize": 16,
                  "height": 48,
                  "marginLeft": 8,
                  "width": "100%",
                },
              ]
            }
            value="Value"
          />
        </View>
      </View>
    </View>
    <View
      style={
        [
          {
            "borderColor": "#2C2C2C",
            "borderRadius": 4,
            "borderStyle": "dashed",
            "borderWidth": 1,
            "marginBottom": 24,
            "marginLeft": 24,
            "marginRight": 24,
            "marginTop": 24,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": 370,
          },
        ]
      }
      width={370}
    >
      <Text
        fontFamily="Rubik-Bold"
        fontSize={16}
        lineHeight={20}
        style={
          [
            {
              "color": "#2C2C2C",
              "fontFamily": "Rubik-Bold",
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "backgroundColor": "#DEDFE0",
              "left": 16,
              "paddingHorizontal": 8,
              "position": "absolute",
              "top": -12,
            },
          ]
        }
      >
        Success input field with text:
      </Text>
      <View
        style={
          [
            {
              "alignItems": "flex-start",
              "justifyContent": "flex-start",
            },
          ]
        }
      >
        <Text
          fontFamily="Rubik-Medium"
          fontSize={12}
          lineHeight={18}
          style={
            [
              {
                "color": "#2C2C2C",
                "fontFamily": "Rubik-Medium",
                "fontSize": 12,
                "lineHeight": 18,
              },
            ]
          }
        >
          success Text input
        </Text>
        <View
          color="rgba(28,197,119,0.5)"
          editable={true}
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "rgba(28,197,119,0.5)",
                "borderRadius": 7,
                "borderWidth": 1,
                "justifyContent": "space-between",
                "minHeight": 48,
                "paddingHorizontal": 8,
                "width": 300,
              },
            ]
          }
          width={300}
        >
          <TextInput
            autoFocus={false}
            editable={true}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder="Placeholder"
            placeholderTextColor="rgba(44,44,44,0.5)"
            style={
              [
                {
                  "color": "#2C2C2C",
                  "fontFamily": "Rubik-Medium",
                  "fontSize": 16,
                  "height": 48,
                  "marginLeft": 8,
                  "width": "100%",
                },
              ]
            }
            value="Success Value"
          />
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      [
        {
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "borderColor": "#2C2C2C",
            "borderRadius": 4,
            "borderStyle": "dashed",
            "borderWidth": 1,
            "marginBottom": 24,
            "marginLeft": 24,
            "marginRight": 24,
            "marginTop": 24,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": 370,
          },
        ]
      }
      width={370}
    >
      <Text
        fontFamily="Rubik-Bold"
        fontSize={16}
        lineHeight={20}
        style={
          [
            {
              "color": "#2C2C2C",
              "fontFamily": "Rubik-Bold",
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "backgroundColor": "#DEDFE0",
              "left": 16,
              "paddingHorizontal": 8,
              "position": "absolute",
              "top": -12,
            },
          ]
        }
      >
        Error input field with text:
      </Text>
      <View
        style={
          [
            {
              "alignItems": "flex-start",
              "justifyContent": "flex-start",
            },
          ]
        }
      >
        <Text
          fontFamily="Rubik-Medium"
          fontSize={12}
          lineHeight={18}
          style={
            [
              {
                "color": "#2C2C2C",
                "fontFamily": "Rubik-Medium",
                "fontSize": 12,
                "lineHeight": 18,
              },
            ]
          }
        >
          Error Text input
        </Text>
        <View
          color="rgba(255,59,59,0.5)"
          editable={true}
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "rgba(255,59,59,0.5)",
                "borderRadius": 7,
                "borderWidth": 1,
                "justifyContent": "space-between",
                "minHeight": 48,
                "paddingHorizontal": 8,
                "width": 300,
              },
            ]
          }
          width={300}
        >
          <TextInput
            autoFocus={false}
            editable={true}
            multiline={false}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder="Placeholder"
            placeholderTextColor="rgba(44,44,44,0.5)"
            style={
              [
                {
                  "color": "#2C2C2C",
                  "fontFamily": "Rubik-Regular",
                  "fontSize": 16,
                  "height": 48,
                  "marginLeft": 8,
                  "width": "100%",
                },
              ]
            }
            value=""
          />
        </View>
        <Text
          fontFamily="Rubik-Regular"
          fontSize={12}
          lineHeight={18}
          style={
            [
              {
                "color": "#2C2C2C",
                "fontFamily": "Rubik-Regular",
                "fontSize": 12,
                "lineHeight": 18,
              },
              {
                "color": "#FF3B3B",
                "marginTop": 4,
              },
            ]
          }
        >
          Oops! Somethings not right here.
        </Text>
      </View>
    </View>
    <View
      style={
        [
          {
            "borderColor": "#2C2C2C",
            "borderRadius": 4,
            "borderStyle": "dashed",
            "borderWidth": 1,
            "marginBottom": 24,
            "marginLeft": 24,
            "marginRight": 24,
            "marginTop": 24,
            "paddingBottom": 16,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingTop": 16,
            "width": 370,
          },
        ]
      }
      width={370}
    >
      <Text
        fontFamily="Rubik-Bold"
        fontSize={16}
        lineHeight={20}
        style={
          [
            {
              "color": "#2C2C2C",
              "fontFamily": "Rubik-Bold",
              "fontSize": 16,
              "lineHeight": 20,
            },
            {
              "backgroundColor": "#DEDFE0",
              "left": 16,
              "paddingHorizontal": 8,
              "position": "absolute",
              "top": -12,
            },
          ]
        }
      >
        Text Box:
      </Text>
      <View
        style={
          [
            {
              "alignItems": "flex-start",
              "justifyContent": "flex-start",
            },
          ]
        }
      >
        <Text
          fontFamily="Rubik-Medium"
          fontSize={12}
          lineHeight={18}
          style={
            [
              {
                "color": "#2C2C2C",
                "fontFamily": "Rubik-Medium",
                "fontSize": 12,
                "lineHeight": 18,
              },
            ]
          }
        >
          Text box
        </Text>
        <View
          color="rgba(199,199,210,0.5)"
          editable={true}
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "#FFFFFF",
                "borderColor": "rgba(199,199,210,0.5)",
                "borderRadius": 7,
                "borderWidth": 1,
                "justifyContent": "space-between",
                "minHeight": 48,
                "paddingHorizontal": 8,
                "width": 300,
              },
            ]
          }
          width={300}
        >
          <TextInput
            autoFocus={false}
            editable={true}
            multiline={true}
            onBlur={[Function]}
            onChangeText={[Function]}
            onFocus={[Function]}
            placeholder="Message"
            placeholderTextColor="rgba(44,44,44,0.5)"
            style={
              [
                {
                  "color": "#2C2C2C",
                  "fontFamily": "Rubik-Regular",
                  "fontSize": 16,
                  "height": 121,
                  "marginLeft": 8,
                  "marginVertical": 8,
                  "width": "100%",
                },
              ]
            }
            value=""
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
